src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2220 @@
+            self.parent.unblank_beam(self.channel)
+
+    def finishScan(self):
+        """
+        Make sure the beam is blanked when the blanker is in 'auto' mode at the end of scanning.
+        """
+        if self.blanker.value is None:
+            self.parent.blank_beam(self.channel)
+
+
+
+class Detector(model.Detector):
+    """
+    This is an extension of model.Detector class. It performs the main functionality
+    of the SEM. It sets up a Dataflow and notifies it every time that an SEM image
+    is captured.
+    """
+
+    def __init__(self, name, role, parent, channel: str, **kwargs):
+        # The acquisition is based on a FSM that roughly looks like this:
+        # Event\State |    Stopped    |   Acquiring    | Receiving data |
+        #    START    | Ready for acq |        .       |       .        |
+        #    DATA     |       .       | Receiving data |       .        |
+        #    STOP     |       .       |     Stopped    |    Stopped     |
+        #    TERM     |     Final     |      Final     |     Final      |
+
+        model.Detector.__init__(self, name, role, parent=parent, **kwargs)
+        self._shape = (256,)  # Depth of the image
+        self.data = SEMDataFlow(self)
+        self.channel = channel
+
+        if self.channel == "electron":
+            self._scanner = self.parent._scanner
+        elif self.channel == "ion":
+            self._scanner = self.parent._fib_scanner
+        else:
+            raise ValueError("Invalid channel name. No scanner available for this channel.")
+
+        brightness_info = self.parent.brightness_info(self.channel)
+        self.brightness = model.FloatContinuous(
+            self.parent.get_brightness(self.channel),
+            brightness_info["range"],
+            unit=brightness_info["unit"],
+            setter=self._setBrightness)
+
+        contrast_info = self.parent.contrast_info(self.channel)
+        self.contrast = model.FloatContinuous(
+            self.parent.get_contrast(self.channel),
+            contrast_info["range"],
+            unit=contrast_info["unit"],
+            setter=self._setContrast)
+
+        detector_type = self.parent.get_detector_type(self.channel)
+        self.detector_type = model.StringEnumerated(
+            detector_type,
+            choices=set(self.parent.detector_type_info(self.channel)["choices"]),
+            setter=self._setDetectorType)
+
+        detector_mode = self.parent.get_detector_mode(self.channel)
+        self.detector_mode = model.StringEnumerated(
+            detector_mode,
+            choices=set(self.parent.detector_mode_info(self.channel)["choices"]),
+            setter=self._setDetectorMode)
+
+        self._genmsg = queue.Queue()  # GEN_*
+        self._generator = None
+
+        # Refresh regularly the values, from the hardware, starting from now
+        self._updateSettings()
+        # self._va_poll = util.RepeatingTimer(5, self._updateSettings, "Settings polling detector")
```

Comment:
Does this need to be uncommented?

Reply:
This causes the tfs ui to change every 5 seconds which causes problems so its disabled for now, see Note below.